ethtxmanager/ethtxmanager.go

```diff
@@ -11,156 +11,612 @@ import (
+			RevertMessage: revertMessage,
+		}
+	}
+
+	result := MonitoredTxResult{
+		ID:     mTx.id,
+		Status: mTx.status,
+		Txs:    txs,
+	}
+
+	return result, nil
+}
+
+// Start will start the tx management, reading txs from storage,
+// send then to the blockchain and keep monitoring them until they
+// get mined
+func (c *Client) Start() {
+	// infinite loop to manage txs as they arrive
+	c.ctx, c.cancel = context.WithCancel(context.Background())
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			return
+		case <-time.After(c.cfg.FrequencyToMonitorTxs.Duration):
+			err := c.monitorTxs(context.Background())
+			if err != nil {
+				c.logErrorAndWait("failed to monitor txs: %v", err)
 			}
 		}
+	}
+}
+
+// Stop will stops the monitored tx management
+func (c *Client) Stop() {
+	c.cancel()
+}
+
+// Reorg updates all monitored txs from provided block number until the last one to
+// Reorged status, allowing it to be reprocessed by the tx monitoring
+func (c *Client) Reorg(ctx context.Context, fromBlockNumber uint64, dbTx pgx.Tx) error {
+	log.Infof("processing reorg from block: %v", fromBlockNumber)
+	mTxs, err := c.storage.GetByBlock(ctx, &fromBlockNumber, nil, dbTx)
+	if err != nil {
+		log.Errorf("failed to monitored tx by block: %v", err)
+		return err
+	}
+	log.Infof("updating %v monitored txs to reorged", len(mTxs))
+	for _, mTx := range mTxs {
+		mTx.blockNumber = nil
+		mTx.status = MonitoredTxStatusReorged
+
+		err = c.storage.Update(ctx, mTx, dbTx)
 		if err != nil {
-			log.Errorf("failed to sequence batches, maximum attempts exceeded, err: %w", err)
-			return fmt.Errorf("failed to sequence batches, maximum attempts exceeded, err: %w", err)
+			log.Errorf("failed to update monitored tx to reorg status: %v", err)
+			return err
 		}
-		// Wait for tx to be mined
-		log.Infof("waiting for tx to be mined. Tx hash: %s, nonce: %d, gasPrice: %d", tx.Hash(), tx.Nonce(), tx.GasPrice().Int64())
-		err = c.ethMan.WaitTxToBeMined(ctx, tx, c.cfg.WaitTxToBeMined.Duration)
-		if err != nil {
-			attempts++
-			if errors.Is(err, runtime.ErrOutOfGas) {
-				gas = increaseGasLimit(tx.Gas(), c.cfg.PercentageToIncreaseGasLimit)
-				log.Infof("out of gas with %d, retrying with %d", tx.Gas(), gas)
+		log.Infof("monitored tx %v status updated to reorged", mTx.id)
+	}
+	log.Infof("reorg from block %v processed successfully", fromBlockNumber)
+	return nil
+}
+
+// monitorTxs process all pending monitored tx
+func (c *Client) monitorTxs(ctx context.Context) error {
+	statusesFilter := []MonitoredTxStatus{MonitoredTxStatusCreated, MonitoredTxStatusSent, MonitoredTxStatusReorged}
+	mTxs, err := c.storage.GetByStatus(ctx, nil, statusesFilter, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get created monitored txs: %v", err)
+	}
+
+	log.Infof("found %v monitored tx to process", len(mTxs))
+
+	for _, mTx := range mTxs {
+		mTx := mTx // force variable shadowing to avoid pointer conflicts
+		mTxLog := log.WithFields("monitored tx", mTx.id)
```

Comment:
@tclemos since logs can now be formatted in JSON, I would suggest camel-case for log fields.